Sparkle/SUAppcast.m

```diff
@@ -115,6 +98,31 @@ - (void)downloadDidFinish:(NSURLDownload *)__unused aDownload
     }
 }
 
+- (NSDictionary *)attributesOfNode:(NSXMLElement *)node
+{
+    NSEnumerator *attributeEnum = [[node attributes] objectEnumerator];
+    NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+
+    for (NSXMLNode *attribute in attributeEnum) {
+        NSString *attrName = [self sparkleNamespacedNameOfNode:attribute];
+        if (!attrName) {
+            continue;
+        }
+        dictionary[attrName] = [attribute stringValue];
+    }
+    return dictionary;
+}
+
+-(NSString *)sparkleNamespacedNameOfNode:(NSXMLNode *)node {
+    // XML namespace prefix is semantically meaningless, so compare namespace URI
+    if ([[node URI] isEqualToString:@"http://www.andymatuschak.org/xml-namespaces/sparkle"]) {
+        NSString *localName = [node localName]; assert(localName);
```

Comment:
Did you mean to leave the assert in, maybe stick it on its own line?

Should https URL be handled here too? I know I may have changed it to https in an appcast for one of my apps. I don't really follow that much with this XML stuff though.